matlabMeas/MeasurementUtilities/initializeSR830Meas2D_Func.m

```diff
@@ -1,73 +1,62 @@
+function [plotHandle] = initializeSR830Meas2D_Func(sweepTypes, starts, stops, deltas)
 
-[time,Real,Imag] = deal(inf);
+% initializeSR830Meas2D_Func({'Freq', 'ST'}, {1000, 0}, {10000, 1}, {1000, 0.01})
 
-timeLabel = "Time [s]";
-voltageAxisName = genSR830Axis(sweepType);
-yLabel = "Current [A]";
+[sweepType1, sweepType2] = sweepTypes{:};
+[start1, start2] = starts{:};
+[stop1, stop2] = stops{:};
+[deltaParam1, deltaParam2] = deltas{:};
 
-subPlotFigure = figure(getNextMATLABFigNum());
-subplot(2,3,1);
-realVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Real",'title',"Real vs Time",'subPlot',1);
+xData = [start2, stop2];
+lenX = length(start2:(2 * (start2 < stop2) - 1) * deltaParam2:stop2);
+yData = [start1, stop1];
+lenY = length(start1:(2 * (start1 < stop1) - 1) *deltaParam1:stop1);
+zData = NaN(lenY, lenX);
 
-subplot(2,3,2)
-imagVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Imag",'title',"Imag vs Time",'subPlot',1);
+yAxisName = genSR830Axis(sweepType1);
+xAxisName = genSR830Axis(sweepType2);
 
-subplot(2,3,3)
-magVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Mag",'title',"Mag vs Time",'subPlot',1);
+myFig = figure(getNextMATLABFigNum());
 
+fig = image(xData, yData, zData, 'CDataMapping', 'scaled');
 
-
-%[vavg1,vavg2,avgxs1,avgxs2,avgys1,avgys2,stdx1,stdx2,stdy1,stdy2,avgmags1,avgmags2,stdm1,stdm2] = deal(inf);
-
-
-subplot(2,3,4)
-if ~doBackAndForth
-    realVsVoltageErr = errorbar(time,Real,Imag,'Bx');
+if start2 < stop2
+    set(gca,'XDir','normal');
 else
-    realVsVoltageErr = errorbar(time,Real,Imag,'Bx');
-    hold on
-    realVsVoltageErr2 = errorbar(time,Real,Imag,'C*');
-    hold off
+    set(gca,'XDir','reverse');
 end
-xlabel(voltageAxisName);
-ylabel(yLabel);
-title("Real vs Voltage");
 
-subplot(2,3,5)
-if ~doBackAndForth
-    imagVsVoltageErr = errorbar(time,Real,Imag,'RX');
+if start1 < stop1
+    set(gca,'YDir','normal');
 else
-    imagVsVoltageErr = errorbar(time,Real,Imag,'RX');
-    hold on
-    imagVsVoltageErr2 = errorbar(time,Real,Imag,'M*');
-    hold off
+    set(gca,'YDir','reverse');
 end
-xlabel(voltageAxisName);
-ylabel(yLabel);
-title("Imag vs Voltage");
 
-subplot(2,3,6)
-if ~doBackAndForth
-    magVsVoltageErr = errorbar(time,Real,Imag,'Kd');
-else
-    magVsVoltageErr = errorbar(time,Real,Imag,'Kd');
-    hold on
-    magVsVoltageErr2 = errorbar(time,Real,Imag,'GO');
-    hold off
-end
-xlabel(voltageAxisName);
-ylabel(yLabel);
-title("Mag vs Voltage");
+colorbar;
 
-if ~doBackAndForth
-    plotHandles = {realVsTime,imagVsTime,magVsTime,realVsVoltageErr,imagVsVoltageErr,magVsVoltageErr};
-else
-    plotHandles = {realVsTime,imagVsTime,magVsTime,realVsVoltageErr,realVsVoltageErr2,imagVsVoltageErr,imagVsVoltageErr2,magVsVoltageErr,magVsVoltageErr2};
-end
-tileFigures(subPlotFigure,1,1,2,[],[0,0,0.5,1]);
+xlabel(xAxisName);
+ylabel(yAxisName);
+
+% Compile Metadata
+figDateFormat = 'mm_dd_yy HH:MM:SS';
```

Comment:
Make a plotData function for 2D sweeps. Any and all plot generation should be done in an initialization script - users should never see the behind the scenes in the actual sweep function. Let the sweep function just do the sweep and nothing else.